syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "custom_types.proto";

service PaymentTemplateService {
  rpc GetAll (GetAllRequest) returns (GetAllPaymentTemplatesResponse);
  rpc GetById (GetPaymentTemplateByIdRequest) returns (PaymentTemplateModel);
}

message PaymentTemplateModel {
  int32 template_id = 1;
  optional int32 customer_id = 2;
  optional string name = 3;
  optional string target_iban = 4;
  optional string default_amount = 5;
}

message GetAllPaymentTemplatesResponse {
  repeated PaymentTemplateModel payment_templates = 1;
}

message GetPaymentTemplateByIdRequest {
  int32 template_id = 1;
}
